pyuvdata/uvbeam.py

```diff
@@ -1817,34 +1815,73 @@ def write_beamfits(self, filename, run_check=True, check_extra=True,
                                     clobber=clobber)
         del(beamfits_obj)
 
-    def read_cst_beam(self, filename, beam_type='power', feed_pol='x', rotate_pol=None,
+    def _read_cst_beam_yaml(self, filename):
+        import yaml
+
+        with open(filename, 'r') as file:
+            settings_dict = yaml.safe_load(file)
+
+        required_keys = ['telescope_name', 'feed_name', 'feed_version',
+                         'model_name', 'model_version', 'history', 'frequencies',
+                         'filenames', 'feed_pol']
+
+        for key in required_keys:
+            if key not in settings_dict:
+                raise ValueError('{key} is a required key in CST settings files '
+                                 'but is not present.'.format(key=key))
+
+        return settings_dict
+
+    def read_cst_beam(self, filename, beam_type='power', feed_pol=None, rotate_pol=None,
                       frequency=None, telescope_name=None, feed_name=None,
                       feed_version=None, model_name=None, model_version=None,
-                      history='', run_check=True, check_extra=True,
+                      history=None, reference_impedance=None, extra_keywords=None,
+                      frequency_select=None, run_check=True, check_extra=True,
                       run_check_acceptability=True):
         """
         Read in data from a cst file.
 
         Args:
-            filename: The cst file or list of files to read from. If a list is passed,
+            filename (str): Either a settings yaml file or a cst text file or
+                list of cst text files to read from. If a list is passed,
                 the files are combined along the appropriate axes.
-            beam_type: what beam_type to read in ('power' or 'efield'). Defaults to 'power'.
-            feed_pol: the feed or polarization or list of feeds or polarizations the files correspond to.
+                Settings yaml files must include the following keywords:
```

Comment:
This is great documentation, but it might be a good idea to have a separate document/memo that defines exactly what goes in these files. Pulling this information out of the docstring/code itself might be too high a barrier to entry for people making the yaml files on the CST simulation side...

Reply:
I agree. I'll work on putting one together. I developed this format in close collaboration with Nicolas Fagnioni, so he's on board, but there should be better documentation.

Reply:
I added a page in the docs giving more details about the settings files. You can see it here: https://pyuvdata.readthedocs.io/en/cst_yaml/cst_settings_yaml.html

When I was checking that it worked on RTD, I discovered that our builds were failing there (I need to figure out how to get pinged when that happens). It was a unicode error that was fixed by having RTD build in python 3.6, so those changes are also in this PR now.

Reply:
Thanks! The new documentation is excellent. Thanks for getting RTD going again too.